player.cpp

```diff
@@ -311,15 +312,17 @@ void test_player() //!OCLINT tests may be long
 {
 #ifndef NDEBUG // no tests in release
 
-  //#define FIX_ISSUE_336
+  #define FIX_ISSUE_336
   #ifdef FIX_ISSUE_336
     {
         player p;
-        cordinate predicted_player_position = predict_players_movement(p);
+        coordinate predicted_player_position = predict_players_movement(p);
         p.move();
-        assert(p.get_position()==predicted_player_position);
-        assert(((predicted_player_position.m_x - get_x(p))<0.001)&&(((predicted_player_position.m_x - get_x(p))>-0.001)));
-        assert(((predicted_player_position.m_x - get_y(p))<0.001)&&(((predicted_player_position.m_x - get_y(p))>-0.001)));
+        assert(p.get_position() == predicted_player_position);
+        assert(predicted_player_position.get_x() - get_x(p) < 0.001
```

Comment:
We should make an issue to standardise tolerance. We can see various 0s before the 1 throughout the code.

Reply:
We should also standardise the method we use this in.

Reply:
To amend @ollyturner 's comment, we should also standardise the method ***of testing whether the error produced is within the tolerance***
Let's make it a separate issue.

Reply:
I agree, that's a very good idea actually. I have made this an issue (#594 ), do you want to pick it :)

Reply:
@TheoPannetier Sure, I will try to implement one 🚀